### PR TITLE
etcdserver: add a test to verify number of MsgAppResp sent is correct.

### PR DESCRIPTION
**Pull request by johncming**

### CODE REVIEW

**Comment by xiang90** [comment]
@johncming 

Can you explain a little bit more about what this test is for? We have both fake transport and fake node, so it basically tests the raft pkg? If that is the case, I think we already cover this in the raft pkg test, no?

**Comment by johncming** [comment]
> @johncming
> 
> Can you explain a little bit more about what this test is for? We have both fake transport and fake node, so it basically tests the raft pkg? If that is the case, I think we already cover this in the raft pkg test, no?

To test the following function：

https://github.com/etcd-io/etcd/blob/6511829d1f3425cf62d5f3744429e81abf1bf733/etcdserver/raft.go#L338

The following code  is covered by testcase show in codecov, but there is no direct test in etcdserver package

https://github.com/etcd-io/etcd/blob/6511829d1f3425cf62d5f3744429e81abf1bf733/etcdserver/raft.go#L345-L351

Whether adding this case is redundant is uncertain and can be ignored if it is not necessary.

thanks.

File: etcdserver/raft_test.go [hunk 1]
```diff
@@ -227,3 +227,44 @@ func TestConfgChangeBlocksApply(t *testing.T) {
 		t.Fatalf("unexpected blocking on execution")
 	}
 }
+
+func TestProcessAppRespMessages(t *testing.T) {
```
Comment (xiang90) [inline etcdserver/raft_test.go]: TestProcessDuplicatedAppRespMessage

**Comment by xiang90** [comment]
@johncming 

Ah. OK. So this tests checks the correctness of the optimization we did (do not send duplicated append responses).

Can you change the test func name to make it clear? thanks!

**Comment by johncming** [comment]
> @johncming
> 
> Ah. OK. So this tests checks the correctness of the optimization we did (do not send duplicated append responses).
> 
> Can you change the test func name to make it clear? thanks!

ok.

**Comment by xiang90** [comment]
@johncming thanks!

**Pull request merged**